### PR TITLE
Issackjohn/create variant

### PR DESCRIPTION
**Pull request by issackjohn**
| Workload 1 | Variant 1 | Variant 2 | uses variant 1 but not enabled anyways |
|--------|--------|--------|--------|
| TodoMVC-JavaScript-ES6-Webpack-Complex-DOM |✅ | -  | - |- | 
| TodoMVC-Lit-Complex-DOMx | - | ✅  | - |- | 
| TodoMVC-Preact-Complex-DOM | - | ✅  |- | 
| TodoMVC-Svelte-Complex-DOM | -  | ✅ |- | 
| TodoMVC-React-Complex-DOM | ✅  | - | - | 
| TodoMVC-Angular-Complex-DOM | -  | ✅ | - | 
| TodoMVC-JavaScript-ES5-Complex-DOM | -  | - | ✅ | 
| TodoMVC-WebComponents-Complex-DOM | -  | - | ✅ | 
| TodoMVC-React-Redux-Complex-DOM | -  | - | ✅ | 
| TodoMVC-Backbone-Complex-DOM | -  | - | ✅ | 
| TodoMVC-Vue-Complex-DOM | -  | - | ✅ | 
| TodoMVC-jQuery-Complex-DOM | -  | - | ✅ | 

Variant two is .tree-area with the property `isolation: isolate` added.

# Hosted at:
https://issackjohn.github.io/Speedometer5

### CODE REVIEW

**Comment by issackjohn** [comment]
I suggest we either remove or implement the other functionality for the other operations `remove` and `modify`.

**Comment by sulekhark** [comment]
> I suggest we either remove the cases or implement the functionality for the operations `remove` and `modify` in the `vary` function inside `createVariant.js`

Yes, I agree. The cases for the operations "remove" and "modify" can be removed. So there will be only one case for "add" and the default case.

Additionally, I think the function "vary" could return the output string and writing to the file can happen outside the function vary.

**Comment by issackjohn** [comment]
@sulekhark PTAL

File: resources/todomvc/big-dom-generator/utils/createVariant.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+const OUTPUT_FILE_PATH = "./dist/big-dom-generator-v2.css";
+let root;
+
+/**
+ * Modifies CSS rules based on the provided selector, tuple, and operation.
+ *
+ * @param {string} cssSelector - The CSS selector to match.
+ * @param {Array<[string, string]>} propValueTuples - The property-value tuples to apply.
+ * @param {string} operation - The operation to perform ('add', 'remove', or 'modify'). At present, only the "add" operation is implemented.
+ * @returns {string} - The modified CSS as a string.
```
Comment (sulekhark) [inline resources/todomvc/big-dom-generator/utils/createVariant.js:15]: This return value no longer holds.

File: resources/todomvc/big-dom-generator/utils/createVariant.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    }
+}
+
+try {
+    const css = fs.readFileSync(INPUT_FILE_PATH, "utf-8");
+    root = postcss.parse(css);
+    vary(".tree-area", [["isolation", "isolate"]], "add");
+    fs.writeFileSync(OUTPUT_FILE_PATH, root.toString());
+} catch (error) {
+    console.error("An error occurred while reading the CSS file:", error);
```
Comment (sulekhark) [inline resources/todomvc/big-dom-generator/utils/createVariant.js:45]: This error message may need to be updated? The error may happen while reading, parsing or writing?

**Comment by sulekhark** [comment]
Looks good! Added two very minor comments.

File: resources/todomvc/big-dom-generator/utils/createVariant.js (hunk 2 above)
Comment (issackjohn) [reply resources/todomvc/big-dom-generator/utils/createVariant.js]: That's correct, done. I just used "An error occurred while processing the CSS:". Hopefully that is okay.

**Review by lpardosixtosMs — COMMENTED** [review]
LGTM overall, my only comment is that it seems structured as something that some can pick up and use to create more variants, but the details to create variant 2 are hard coded in `createVariant.js`.  I think it would look better if `createVariant.js` contains the logic only and the function is called from somewhere else.

**Comment by sulekhark** [comment]
> LGTM overall, my only comment is that it seems structured as something that some can pick up and use to create more variants, but the details to create variant 2 are hard coded in `createVariant.js`. I think it would look better if `createVariant.js` contains the logic only and the function is called from somewhere else.

We decided to make the function `vary` general (i.e., can be invoked several times to make multiple changes), but keep the invocation of function `vary` within the same file to keep things simple for the present requirement, which is just one change. If we do need to create more variants or more variance in future, then the above suggested separation can be an easy improvement to generalization. For the present, we thought it best to keep things simple.

**Comment by lpardosixtosMs** [comment]
> > LGTM overall, my only comment is that it seems structured as something that some can pick up and use to create more variants, but the details to create variant 2 are hard coded in `createVariant.js`. I think it would look better if `createVariant.js` contains the logic only and the function is called from somewhere else.
> 
> We decided to make the function `vary` general (i.e., can be invoked several times to make multiple changes), but keep the invocation of function `vary` within the same file to keep things simple for the present requirement, which is just one change. If we do need to create more variants or more variance in future, then the above suggested separation can be an easy improvement towards more generalization. For the present, we thought it best to keep things simple.

SGTM

**Review by lpardosixtosMs — APPROVED** [review]

**Pull request merged**